lib/puppet/util/run_mode.rb

```diff
@@ -55,11 +55,19 @@ def which_dir( system, user )
 
     class UnixRunMode < RunMode
       def conf_dir
-        which_dir("/etc/puppet", "~/.puppet")
+        which_dir("/etc/puppetlabs/agent", "~/.puppet")
       end
 
       def var_dir
-        which_dir("/var/lib/puppet", "~/.puppet/var")
+        which_dir("/opt/puppetlabs/agent/cache", "~/.puppet/var")
+      end
+
+      def run_dir
+        which_dir("/var/run/puppetlabs", "~/.puppet/var/run")
+      end
+
+      def log_dir
+        which_dir("/var/log/puppetlabs/agent", "~/.puppet/var/log/")
```

Comment:
Played around with this locally a bit and I'm having success with the following:
```ruby
which_dir("/var/log/puppetlabs/agent", Puppet[:vardir] + "/log")
```
Spec tests pass except for the run_mode_spec which would be updated accordingly.